<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BatchProcessor</class>
 <widget class="QDialog" name="BatchProcessor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>547</width>
    <height>377</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QModBus Batch Processor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="gridLayout" columnstretch="0,1,10,0,0">
     <item row="1" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Period (in seconds)</string>
       </property>
       <property name="buddy">
        <cstring>intervalSpinBox</cstring>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="intervalSpinBox">
       <property name="toolTip">
        <string>Use 0 for single-shot execution.</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item row="1" column="2" colspan="2">
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Output file</string>
       </property>
       <property name="buddy">
        <cstring>outputFileEdit</cstring>
       </property>
      </widget>
     </item>
     <item row="3" column="1" colspan="3">
      <widget class="QLineEdit" name="outputFileEdit">
       <property name="toolTip">
        <string>If empty, the output is only logged below.</string>
       </property>
      </widget>
     </item>
     <item row="3" column="4">
      <widget class="QToolButton" name="openOutputFileButton">
       <property name="text">
        <string>...</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1" colspan="3">
      <widget class="QPlainTextEdit" name="batchEdit">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Batch is a sequence of commands separated by the semicolon: &lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;cmd&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;0&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;;&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;cmd&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;1&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;;...;&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;cmd&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;m&lt;/span&gt;&lt;/p&gt;&lt;p&gt;One command &lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;cmd&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;i&lt;/span&gt; has the following syntax: &lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;slave&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;i&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;x&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;func&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;i&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;:&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;data&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;i0&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;,&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;data&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;i1&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;,...,&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;data&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;in&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;slave&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;i&lt;/span&gt;: Slave ID in &lt;span style=&quot; text-decoration: underline;&quot;&gt;dec&lt;/span&gt;.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;func&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;i&lt;/span&gt;: Function code in &lt;span style=&quot; text-decoration: underline;&quot;&gt;hex&lt;/span&gt; (01=Read Coils, 02=Read Discrete Inputs, 03=Read Holding Registers, 04=Read Input Registers, 05=Write Single Coil, 06=Write Single Register, 0F=Write Multiple Coils, 10=Write Multiple Registers).&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;data&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000; vertical-align:sub;&quot;&gt;ij&lt;/span&gt;: Either &lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;addr&lt;/span&gt; (for all &lt;span style=&quot; font-style:italic;&quot;&gt;Read&lt;/span&gt; commands) or &lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;addr&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;=&lt;/span&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;val&lt;/span&gt; (for all &lt;span style=&quot; font-style:italic;&quot;&gt;Write&lt;/span&gt; commands), where &lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;addr&lt;/span&gt; is the coil/input/register address in &lt;span style=&quot; text-decoration: underline;&quot;&gt;dec&lt;/span&gt; and &lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;val&lt;/span&gt; is the new value in &lt;span style=&quot; text-decoration: underline;&quot;&gt;dec&lt;/span&gt;.&lt;/p&gt;&lt;p&gt;For example, &lt;span style=&quot; font-weight:600;&quot;&gt;13x10:7=2,12=1;62x02:8,9,10,11,12&lt;/span&gt; first writes 2 to the holding register 7 and 1 to the holding register 12 of the slave 13, then it reads discrete inputs 8-12 from slave 62.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="lineWrapMode">
        <enum>QPlainTextEdit::NoWrap</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="batchLbl">
       <property name="text">
        <string>Batch</string>
       </property>
       <property name="buddy">
        <cstring>batchEdit</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="4">
      <widget class="QToolButton" name="batchLoadBtn">
       <property name="text">
        <string>...</string>
       </property>
       <property name="popupMode">
        <enum>QToolButton::MenuButtonPopup</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="startButton">
       <property name="text">
        <string>Start</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="stopButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Stop</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_8">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="closeButton">
       <property name="text">
        <string>Close</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="lblLog">
     <property name="text">
      <string>Output Log:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPlainTextEdit" name="txtLog">
     <property name="lineWrapMode">
      <enum>QPlainTextEdit::NoWrap</enum>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>batchEdit</tabstop>
  <tabstop>intervalSpinBox</tabstop>
  <tabstop>outputFileEdit</tabstop>
  <tabstop>openOutputFileButton</tabstop>
  <tabstop>startButton</tabstop>
  <tabstop>stopButton</tabstop>
  <tabstop>closeButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>closeButton</sender>
   <signal>clicked()</signal>
   <receiver>BatchProcessor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>541</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>99</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openOutputFileButton</sender>
   <signal>clicked()</signal>
   <receiver>BatchProcessor</receiver>
   <slot>browseOutputFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>541</x>
     <y>188</y>
    </hint>
    <hint type="destinationlabel">
     <x>505</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>BatchProcessor</receiver>
   <slot>start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>67</x>
     <y>144</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopButton</sender>
   <signal>clicked()</signal>
   <receiver>BatchProcessor</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>batchLoadBtn</sender>
   <signal>clicked()</signal>
   <receiver>BatchProcessor</receiver>
   <slot>browseBatchFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>528</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>54</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>start()</slot>
  <slot>stop()</slot>
  <slot>browseOutputFile()</slot>
  <slot>browseBatchFile()</slot>
 </slots>
</ui>
